p2p/host/autorelay/relay_finder.go

```diff
@@ -59,7 +59,7 @@ type relayFinder struct {
 	ctxCancel   context.CancelFunc
 	ctxCancelMx sync.Mutex
 
-	peerSource func(int) <-chan peer.AddrInfo
+	peerSource func(context.Context, int) <-chan peer.AddrInfo
```

Comment:
Would it make sense to remove the `int` here, and have the caller give us new nodes until we cancel the context?

Reply:
Maybe ?
The `int` allows to create a buffered channel of the right size for synchronous or partially synchronous implementations tho.